packages/common-tests/src/wait-utils.ts

```diff
@@ -0,0 +1,14 @@
+import { screen, waitFor } from '@testing-library/react';
+export const waitForLoadingCompleted = async (): Promise<void> => {
+  await waitFor(
+    () => expect(screen.queryAllByTestId('loading-spinner')).toHaveLength(0),
+    { timeout: 20_000 }
```

Comment:
Does this mean 20 seconds timeout? We talked about this earlier in another pull request also. In my opinion, if we need to wait for **a mocked solution** longer than 2 seconds (where the 2 seconds is a hard coded debounce value or something, set in purpose), there is something wrong in the mock, in the test env or in the feature itself. When we have mocks, we are working in a zero latency environment, so we should never need to wait this long. The whole page full of components should load in a couple of seconds, so why would 1 single component load longer?

However, it is a good thing that we have a global utility for this loading-wait. It will make the implementations and the tests more consistent.

Reply:
yes that's kind a different problem to solve. Now I just wanted to make the tests pass more robust way.